bin/publish-modules

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env node
+
+var exec = require( 'child-process-promise' ).exec;
+var semver = require( 'semver' );
+
+console.log( 'Modules requiring publish:' );
+
+function getAllModules() {
+	var command = 'find client -type f -print | grep \'package.json\'';
+	return exec( command ).then( function( result ) {
+		return result.stdout.split( /\s+/ );
+	} );
+}
+
+function getPublishableModules( allModules ) {
+	var packageJson;
+	return allModules.filter( function( modulePath ) {
+		if ( !modulePath ) {
+			return;
+		}
+		packageJson = require( modulePath );
+		if ( !packageJson.private ) {
+			return true;
+		}
+	} );
+}
+
+function publishModule( modulePath ) {
+	return exec( 'make -C ' + modulePath.slice( 0, -12 ) + ' publish' )
+		.progress( function( childProcess ) {
+			childProcess.stdout.on( 'data', function( data ) {
+				console.log( data.toString() );
+			} );
+			childProcess.stderr.on( 'data', function( data ) {
+				console.log( data.toString() );
+			} );
+		} )
+}
+
+function publishModules() {
+	getAllModules()
+		.then( getPublishableModules )
+		.then( requiresUpdate );
+}
+
+function requiresUpdate( modules ) {
+	var modulePath;
+	var packageJson;
+	if ( !modules.length ) {
+		throw true;
+	}
+	modulePath = modules.shift();
+	packageJson = require( modulePath );
+
+	exec( 'npm view ' + packageJson.name + ' version' )
```

Comment:
I think `npm` has an API, that would be the safest thing to use I think but it might make the code more verbose